okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -0,0 +1,827 @@
+ * if you are writing a application that needs to aggressively release unused resources you may do
+ * so.
+ *
+ * Shutdown the dispatcher's executor service with [shutdown()][ExecutorService.shutdown]. This will
+ * also cause future calls to the client to be rejected.
+ *
+ * ```
+ * client.dispatcher().executorService().shutdown();
+ * ```
+ *
+ * Clear the connection pool with [evictAll()][ConnectionPool.evictAll]. Note that the connection
+ * pool's daemon thread may not exit immediately.
+ *
+ * ```
+ * client.connectionPool().evictAll();
+ * ```
+ *
+ * If your client has a cache, call [close()][Cache.close]. Note that it is an error to create calls
+ * against a cache that is closed, and doing so will cause the call to crash.
+ *
+ * ```
+ * client.cache().close();
+ * ```
+ *
+ * OkHttp also uses daemon threads for HTTP/2 connections. These will exit automatically if they
+ * remain idle.
+ */
+open class OkHttpClient internal constructor(
+  builder: Builder
+) : Cloneable, Call.Factory, WebSocket.Factory {
+  private val dispatcher: Dispatcher = builder.dispatcher
+  private val proxy: Proxy? = builder.proxy
+  private val protocols: List<Protocol> = builder.protocols
+  private val connectionSpecs: List<ConnectionSpec> = builder.connectionSpecs
+  private val interceptors: List<Interceptor> =
+      Util.immutableList(builder.interceptors)
+  private val networkInterceptors: List<Interceptor> =
+      Util.immutableList(builder.networkInterceptors)
+  private val eventListenerFactory: EventListener.Factory = builder.eventListenerFactory
+  private val proxySelector: ProxySelector = builder.proxySelector
+  private val cookieJar: CookieJar = builder.cookieJar
+  private val cache: Cache? = builder.cache
+  private val socketFactory: SocketFactory = builder.socketFactory
+  private val sslSocketFactory: SSLSocketFactory?
+  private val hostnameVerifier: HostnameVerifier = builder.hostnameVerifier
+  private val certificatePinner: CertificatePinner
+  private val proxyAuthenticator: Authenticator = builder.proxyAuthenticator
+  private val authenticator: Authenticator = builder.authenticator
+  private val connectionPool: ConnectionPool = builder.connectionPool
+  private val dns: Dns = builder.dns
+  private val followSslRedirects: Boolean = builder.followSslRedirects
+  private val followRedirects: Boolean = builder.followRedirects
+  private val retryOnConnectionFailure: Boolean = builder.retryOnConnectionFailure
+  private val callTimeout: Int = builder.callTimeout
+  private val connectTimeout: Int = builder.connectTimeout
+  private val readTimeout: Int = builder.readTimeout
+  private val writeTimeout: Int = builder.writeTimeout
+  private val pingInterval: Int = builder.pingInterval
+  private val internalCache: InternalCache? = builder.internalCache
+  private val certificateChainCleaner: CertificateChainCleaner?
+
+  constructor() : this(Builder())
+
+  init {
+    var isTLS = false
+    for (spec in connectionSpecs) {
```

Comment:
Is this more idiomatic?

```
isTLS = connectionSpecs.any { it.isTLS }
```

Reply:
Yes! I wanna save for follow up. Would be excellent for IntelliJ to quick fix this for me.